ff_derive/src/lib.rs

```diff
@@ -1233,7 +1233,9 @@ fn prime_field_impl(
                     };
 
                     // Mask away the unused most-significant bits.
-                    tmp.0.as_mut()[#top_limb_index] &= 0xffffffffffffffff >> REPR_SHAVE_BITS;
+                    // Note: In some edge cases, `REPR_SHAVE_BITS` could be 64, in which case
+                    // `0xfff... >> REPR_SHAVE_BITS` overflows. So use `checked_shr` instead.
```

Comment:
```suggestion
                    // Note: In some edge cases, `REPR_SHAVE_BITS` could be 64, in which case
                    // `0xfff... >> REPR_SHAVE_BITS` overflows. So use `checked_shr` instead.
                    // This is always sufficient because we will have at most one spare limb
                    // to accommodate values of up to twice the modulus.
```